src/wazuh_db/wdb_global.c

```diff
@@ -1254,28 +1257,41 @@ wdbc_result wdb_global_unassign_agent_group(wdb_t *wdb, int id, cJSON* j_groups)
     cJSON* j_group_name = NULL;
     wdbc_result result = WDBC_OK;
     cJSON_ArrayForEach (j_group_name, j_groups) {
-        if (cJSON_IsString(j_group_name)){
+        if (cJSON_IsString(j_group_name)) {
             char* group_name = j_group_name->valuestring;
             cJSON* j_find_response = wdb_global_find_group(wdb, group_name);
             if (j_find_response) {
-                cJSON* j_group_id = cJSON_GetObjectItem(j_find_response->child,"id");
-                int group_id = j_group_id->valueint;
+                cJSON* j_group_id = cJSON_GetObjectItem(j_find_response->child, "id");
                 cJSON_Delete(j_find_response);
-                if (OS_INVALID == wdb_global_delete_tuple_belong(wdb, group_id, id)) {
-                    mdebug1("Unable to delete group '%s' for agent '%d'", group_name, id);
-                    result = WDBC_ERROR;
+                if (cJSON_IsNumber(j_group_id)) {
+                    if (OS_SUCCESS == wdb_global_delete_tuple_belong(wdb, j_group_id->valueint, id)) {
+                        if (OS_INVALID == wdb_global_get_agent_max_group_priority(wdb, id)) {
+                            cJSON* j_default_group = cJSON_CreateArray();
+                            cJSON_AddItemToArray(j_default_group, cJSON_CreateString("default"));
+                            if (WDBC_OK == wdb_global_assign_agent_group(wdb, id, j_default_group, 0)) {
+                                mdebug1("Agent '%03d' reassigned to 'default' group", id);
+                            } else {
+                                merror("There was an error assigning the agent '%03d' to default group", id);
```

Comment:
Also as we discussed, we keep it as a warning since it doesn't change the result variable to an error value and doesn't interrupt the correct execution of this function.